src/tracker-component.js

```diff
@@ -0,0 +1,102 @@
+/* @flow */
+
+import { getClientID, getMerchantID } from '@paypal/sdk-client/src';
+
+type TrackingType = 'view' | 'cartEvent' | 'purchase';
+
+type CartEventType = 'addToCart' | 'setCart' | 'removeFromCart';
+
+type Environment = 'stage' | 'sandbox' | 'production';
+
+type Product = {|
+    id : string,
+    url : string,
+    sku? : string,
+    name? : string,
+    description? : string,
+    imgUrl? : string,
+    otherImages? : $ReadOnlyArray<string>
+|};
+
+type Cart = {|
+    cartId? : string,
+    items : $ReadOnlyArray<Product>,
+    emailCampaignId? : string,
+    price? : number,
+    currencyCode? : string,
+    keywords? : $ReadOnlyArray<string>
+|};
+
+type RemoveCart = {|
+    cartId? : string,
+    items : $ReadOnlyArray<{ id : string }>
+|};
+
+type ParamsToBeaconUrl = ({
+    trackingType : TrackingType,
+    data : Cart | RemoveCart | { cartId : string } | { pageUrl : string }
+}) => string;
+
+type Config = {|
+    user : {
+        id : string,
+        email? : string, // mandatory if unbranded cart recovery
+        name? : string
+    },
+    property? : {
+        id : string
+    },
+    paramsToBeaconUrl? : ParamsToBeaconUrl,
+    env? : Environment
+|};
+
+const track = <T>(config : Config, trackingType : TrackingType, trackingData : T) : Promise<void> => {
+    const encodeData = data => encodeURIComponent(btoa(JSON.stringify(data)));
+
+    return new Promise((resolve, reject) => {
+        const img = document.createElement('img');
+        img.style.display = 'none';
+
+        const data = {
+            ...trackingData,
+            user:       config.user,
+            property:   config.property,
+            trackingType,
+            clientId:   getClientID(),
+            merchantId: getMerchantID()
+        };
+
+        if (config.paramsToBeaconUrl) {
```

Comment:
Can you add a comment here that describes the purpose of paramsToBeaconUrl?

Reply:
Done